### PR TITLE
Error : Keys.SPAWNABLE_ENTITY_TYPE

### PR DESCRIPTION
**Pull request by rexbut**
Spawn_Egg doesn't use item damage in 1.10.2

### CODE REVIEW

File: src/main/java/org/spongepowered/common/data/processor/data/item/SpawnableDataProcessor.java [hunk 1]
```diff
@@ -55,20 +58,24 @@ public SpawnableDataProcessor() {
...
     @Override
     public Optional<EntityType> getVal(ItemStack itemStack) {
-        final Class<?> entity = EntityList.getClassFromID(itemStack.getItemDamage());
-        for (EntityType type : SpongeImpl.getRegistry().getAllOf(EntityType.class)) {
-            if (type.getEntityClass().equals(entity)) {
-                return Optional.of(type);
+        final String name = ItemMonsterPlacer.getEntityIdFromItem(itemStack);
+        if (name != null) {
+            final Class<?> entity = EntityList.NAME_TO_CLASS.get(name);
+            for (EntityType type : SpongeImpl.getRegistry().getAllOf(EntityType.class)) {
```
Comment (Aaron1011) [inline src/main/java/org/spongepowered/common/data/processor/data/item/SpawnableDataProcessor.java]: Can you change this to `EntityTypeRegistryModule.getInstance()#getForClass`?

**Comment by Aaron1011** [comment]
You also need to update the `removeFrom` method.

File: src/main/java/org/spongepowered/common/data/processor/data/item/SpawnableDataProcessor.java [hunk 2]
```diff
@@ -94,16 +95,20 @@ public SpawnableData createManipulator() {
...
-            ItemStack stack = (ItemStack) container;
-            Optional<EntityType> old = getVal(stack);
-            if (!old.isPresent()) {
-                return DataTransactionResult.successNoData();
-            }
-            stack.setItemDamage(0);
-            return DataTransactionResult.successRemove(constructImmutableValue(old.get()));
-        }
-        return DataTransactionResult.failNoData();
+		if (!supports(container)) {
```
Comment (JBYoshi) [inline src/main/java/org/spongepowered/common/data/processor/data/item/SpawnableDataProcessor.java]: Use spaces instead of tabs.

Comment (rexbut) [inline src/main/java/org/spongepowered/common/data/processor/data/item/SpawnableDataProcessor.java]: The problem has been corrected

**Comment by Aaron1011** [comment]
@rexbut: Can you squash these commits into one?

**Comment by rexbut** [comment]
@Aaron1011: It is done

**Comment by Aaron1011** [comment]
Closed in https://github.com/SpongePowered/SpongeCommon/commit/10c728cc553ce6f565472c88e04a169f12287f3c.

**Pull request closed without merging**